core/src/main/scala/org/bitcoins/core/wallet/utxo/AddressTag.scala

```diff
@@ -0,0 +1,14 @@
+package org.bitcoins.core.wallet.utxo
+
+trait AddressTag {
+  val typeName: String
+}
+
+trait AddressTagFactory[TagType <: AddressTag] {
+
+  val all: Vector[TagType]
```

Comment:
Don't use `val` in `trait` if you can avoid it, it can cause weird initialization errors with Scala. Also you don't want to be purposeful about using `val` in traits -- do you want all inheritors not be able to define this as a method? Maye i want to read all tag types from a database?

https://stackoverflow.com/a/19642301